### PR TITLE
Add spark-rapids-ml PCA algo example

### PR DESCRIPTION
**Pull request by wjxiz1992**
- add README
- add example scala code for spark-shell
- add notebook for quick demo
- add dockerfile for quick test
- add example app scala code for spark-submit

Signed-off-by: Allen Xu <allxu@nvidia.com>

### CODE REVIEW

**Comment by wjxiz1992** [comment]
Hi @GaryShen2008 @nvliyuan I've verified this PCA example on our VR machines. Please help review as the real VR will need the content in this PR.

File: examples/pca/PCA example notebook.ipynb [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+       "pca_f193f5312c9c"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "val pcaGpu = new com.nvidia.spark.ml.feature.PCA().setInputCol(\"feature\").setOutputCol(\"pca_features\").setK(3).setUseGemm(true).setUseCuSolverSVD(true)"
```
Comment (GaryShen2008) [inline examples/pca/PCA example notebook.ipynb:201]: should we have the default value, so no need to write "setUseGemm(true).setUseCuSolverSVD(true)"

File: examples/pca/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,67 @@
+ARG CUDA_VER=11.2.2
```
Comment (GaryShen2008) [inline examples/pca/Dockerfile:1]: Need copyright

File: examples/pca/spark-env.sh [hunk 3]
```diff
@@ -0,0 +1,2 @@
+SPARK_MASTER_HOST=127.0.0.1
```
Comment (GaryShen2008) [inline examples/pca/spark-env.sh:1]: copyright?

File: examples/pca/spark-submit.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
+$SPARK_HOME/bin/spark-submit \
```
Comment (GaryShen2008) [inline examples/pca/spark-submit.sh:1]: copyright?

File: examples/pca/start-spark.sh [hunk 5]
```diff
@@ -0,0 +1,2 @@
+/opt/spark-3.1.2-bin-hadoop3.2/sbin/start-master.sh
```
Comment (GaryShen2008) [inline examples/pca/start-spark.sh:1]: copyright?

Comment (wjxiz1992) [reply examples/pca/start-spark.sh:18]: Add copyright statement to those files.

**Review by GaryShen2008 — APPROVED** [review]

**Pull request merged**